core/pkg/eval/ievaluator.go

```diff
@@ -2,25 +2,71 @@ package eval
 
 import (
 	"github.com/open-feature/flagd/core/pkg/sync"
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.18.0"
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
+const (
+	FlagdProviderName        = "flagd"
+	FeatureFlagReasonKeyName = "feature_flag.reason"
+	ExceptionTypeKeyName     = "exception.type"
+	ExceptionTypeKey         = attribute.Key(ExceptionTypeKeyName)
+	FeatureFlagReasonKey     = attribute.Key(FeatureFlagReasonKeyName)
+)
+
 type AnyValue struct {
 	Value   interface{}
 	Variant string
 	Reason  string
 	FlagKey string
+	Error   error
 }
 
-func NewAnyValue(value interface{}, variant string, reason string, flagKey string) AnyValue {
+func NewAnyValue(value interface{}, variant string, reason string, flagKey string, err error) AnyValue {
 	return AnyValue{
 		Value:   value,
 		Variant: variant,
 		Reason:  reason,
 		FlagKey: flagKey,
+		Error:   err,
+	}
+}
+
+func (a AnyValue) GetImpressionAttributes() []attribute.KeyValue {
+	return []attribute.KeyValue{
+		semconv.FeatureFlagProviderName(FlagdProviderName),
+		semconv.FeatureFlagKey(a.FlagKey),
+		semconv.FeatureFlagVariant(a.Variant),
+		FeatureFlagReason(a.Reason),
 	}
 }
```

Comment:
Idk if we should add information about semantic convention directly in the "interface". 
I would rather have semantic conventions in the `metrics.go` or a new file under that package. 
I am thinking about the use case where we could decide to add/remove a dimension from a metric. To implement this, we would need to know which object implements these attributes instead of looking directly nearby where we register the metric datapoint.

Reply:
resolved